core/src/main/scala/org/apache/spark/resource/ExecutorResourceRequests.scala

```diff
@@ -132,6 +144,14 @@ class ExecutorResourceRequests() extends Serializable {
     this
   }
 
+  /**
+   * Add a certain [[ExecutorResourceRequest]] to the request set.
+   */
+  def addRequest(treq: ExecutorResourceRequest): this.type = {
```

Comment:
Add this as same to what `TaskResourceRequest` does.

Reply:
Actually, I'm thinking do we really need to expose  `TaskResourceRequest` as a public API? As users can always add requests from the above convenient APIs. cc @tgravescs